presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkNativeTaskInputs.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrestoSparkNativeTaskInputs
+        implements PrestoSparkTaskInputs
+{
+    // fragmentId -> PrestoSparkShuffleReadDescriptor for native execution
+    private final Map<String, PrestoSparkShuffleReadDescriptor> shuffleReadDescriptors;
```

Comment:
Why is one descriptor a map and other one just optional ?

Reply:
Because there is 0 or 1 write descriptor for a fragment plan, but 0 or multiple (1, 2 or more) read descriptor. 
Using map for read can represent all counts and using map for write does not make much sense as there is at most 1. Plus using map for write may make people think there might be more than 1 write descriptor, which is semantically wrong. That is why we chose to have Option wrap around the write